flink-formats/flink-format-common/src/main/java/org/apache/flink/formats/common/Converter.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.formats.common;
+
+import java.io.Serializable;
+
+/**
+ * A generic interface for converting data types.
+ *
+ * @param <From> The type of the element to be converted.
+ * @param <To> The output type.
+ * @param <C> The context for passing optional conversion instructions.
+ */
+public interface Converter<From, To, C> extends Serializable {
```

Comment:
I guess this interface is supposed to be `@PublicEvolving` or `@Experimental` since it might be used in the future by a lot of formats.

Reply:
Good point, is there a preference which one to use in this particular case?

Reply:
I think you have to mark it is as `@PublicEvolving` because it is used by your new CSV format that also is `@PublicEvolving`